######### Objective ############
We have 2 fields, our objective is to compare the yield between on different samples. (Using T value method)
t - value = (Signal/Noise) = (Difference between the means/variability of group)

Formula = |X1-X2|/Sqrt((S1^2/n1)+(S2^2/n2))

### T test is applied when

1) population Standard deviation is not available
2) n is less than 30


In [19]:
import pandas as pd
import numpy as np
import math

In [4]:
## importing Data Set. -- Each data point is a yield from field (different samples).
df=pd.read_excel("Field.xlsx")
df

Field_1  Field_2
0      15.2     15.9
1      15.3     15.9
2      16.0     15.2
3      15.8     16.6
4      15.6     15.2
5      14.9     15.8
6      15.0     15.8
7      15.4     16.2
8      15.6     15.6
9      15.7     15.6
10     15.5     15.8
11     15.2     15.5
12     15.5     15.5
13     15.1     15.5
14     15.3     14.9
15     15.0     15.9

In [9]:
F1_mean = np.mean(df.Field_1)
F2_mean = np.mean(df.Field_2)

print(" Mean value of field 1 is {0} \n Mean value of field 2 is {1} ".format(F1_mean,F2_mean))


 Mean value of field 1 is 15.38125 
 Mean value of field 2 is 15.68125 


In [12]:
F1_std = np.std(df.Field_1)
F2_std = np.std(df.Field_2)

print(" Standerd deviation of field 1 is {0} \n Standerd deviation of field 2 is {1} ".format(F1_std,F2_std))


 Standerd deviation of field 1 is 0.3025284077570237 
 Standerd deviation of field 2 is 0.39404750665370314 


In [15]:
F1_var = F1_std*F1_std
F2_var = F2_std*F2_std
print(" Variance of field 1 is {0} \n Variance of field 2 is {1} ".format(F1_var,F2_var))

 Variance of field 1 is 0.09152343750000001 
 Variance of field 2 is 0.15527343750000022 


In [17]:
n1=len(df.Field_1)
n2=len(df.Field_2)

In [21]:
## Final Calculation

t_value = abs(F1_mean-F2_mean)/math.sqrt((F1_var/n1)+(F2_var/n2))
print("T - value is ",t_value)



T - value is  2.4155243395084756


H0 (Null hypothesis) - No Statistical difference between the samples.
H1 (Alternate Hypothesis) - Exists a statistical difference betwen the samples.

If our t-value is less than critical -- fail to reject the null hypothesis.
If our t-value is more than critical-- reject the null hypothesis.

## finding the critical value
1) refer table for Df at rows and columns as significance (P value)

in our case we have 32 data points, 2 variable. So df is 32-2=30.
Mostly used/ tested signifcance is .05. our case is two tailed and critical value is 2.04.
Calculated T value is 2.4 which is higher and reject hypothesis.

Conclusion
"There exists a statistic difference"
